### PR TITLE
Pull request to address issue #27

### PR DESCRIPTION
**Pull request by ved-rivos**
1. 3 capability bits added to enumerate PD20, PD17, PD8
2. Device context configuration checks updated to check supported PDTP mode
3. If PSCID wider than supported process_id width then behavior is UNSPECIFIED

### CODE REVIEW

File: iommu_registers.adoc [hunk 1]
```diff
@@ -236,10 +239,18 @@ the IOMMU. At reset, the register shall contain the IOMMU supported features.
...
-|63:48 |_custom_   |RO        | _Reserved for custom use_
+|38    |`PD20`     |RO        | Three level PDT with 20-bit process_id supported.
+|39    |`PD17`     |RO        | Two level PDT with 17-bit process_id supported.
+|40    |`PD8`      |RO        | One level PDT with 8-bit process_id supported.
+|55:41 | reserved  |RO        | Reserved for standard use
+|63:56 |_custom_   |RO        | _Reserved for custom use_
 |===
 
+The implementation must support a `PSCID` that is at least as wide as the 
+supported `process_id` width. The behavior of configuring a `PSCID` in in-memory
```
Comment (ved-rivos) [inline iommu_registers.adoc:250]: The implementation must support a `PSCID` that is at least as wide as the widest supported `process_id` in the system.

File: iommu_registers.adoc [hunk 2]
```diff
@@ -236,8 +239,11 @@ the IOMMU. At reset, the register shall contain the IOMMU supported features.
 |30    |`HPM`     |RO         | IOMMU implements a hardware performance monitor.
 |31    |`DBG`      |RO        | IOMMU supports the translation-request interface
 |37:32 |`PAS`      |RO        | Physical Address Size (value between 32 and 56)
-|47:38 | reserved  |RO        | Reserved for standard use
-|63:48 |_custom_   |RO        | _Reserved for custom use_
+|38    |`PD20`     |RO        | Three level PDT with 20-bit process_id supported.
```
Comment (pperesse) [inline iommu_registers.adoc:242]: You may want to reorder here as well PD8, PD17 and PD20 to bits 38, 39 and 40

Comment (ved-rivos) [reply iommu_registers.adoc]: Done.

**Comment by pperesse** [comment]
@ved-rivos In the register layout can you modify:
Offset 624, Reserved, Size 64 (instead of 58)
Offset 688 (instead of 682), Custom, size 72 (instead of 78)
Thanks

**Comment by ved-rivos** [comment]
Updated.

**Pull request merged**